nixos/modules/system/boot/resolved.nix

```diff
@@ -48,6 +48,20 @@ in
       '';
     };
 
+    services.resolved.mdns = mkOption {
+      default = "true";
+      example = "false";
```

Comment:
```suggestion
      default = "false";
      example = "true";
```

Reply:
I believe that you're suggesting this, because enabling mdns by default would enable resolving domains that wouldn't have resolved previously without deliberate change by a nixos user, correct?
Do you happen to know why LLMNR is enabled by default, when that seems to be the rarer technology nowadays, compared to mdns (see [an article by someone who seems to be part of MS's Windows Core Networking team](https://techcommunity.microsoft.com/t5/networking-blog/aligning-on-mdns-ramping-down-netbios-name-resolution-and-llmnr/ba-p/3290816))?

Reply:
I've changed the default to `false` as you suggested with 357c2f3548087112b33eb175bd8d074492a418f4

Reply:
MulticastDNS is enabled by default by [systemd](https://github.com/systemd/systemd/blob/d84bdadb434c1c07a9dd548f757ada0e2909262f/meson_options.txt#L324-L327) (so this module needs to set `default = true` for compatibility).

However, that's only the global option. There's also a [per-network setting](https://www.freedesktop.org/software/systemd/man/systemd.network.html#MulticastDNS=), which defaults to false, that needs to be enabled as well.

`resolvectl` output on a vanilla nixos vm with `services.resolved.enable = true`:
<details>

```
Global
           Protocols: +LLMNR +mDNS -DNSOverTLS DNSSEC=allow-downgrade/supported
    resolv.conf mode: stub
Fallback DNS Servers: 1.1.1.1#cloudflare-dns.com 8.8.8.8#dns.google
                      1.0.0.1#cloudflare-dns.com 8.8.4.4#dns.google
                      2606:4700:4700::1111#cloudflare-dns.com
                      2001:4860:4860::8888#dns.google
                      2606:4700:4700::1001#cloudflare-dns.com
                      2001:4860:4860::8844#dns.google

Link 2 (eth0)
Current Scopes: LLMNR/IPv4 LLMNR/IPv6
     Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS
                DNSSEC=allow-downgrade/supported
```

</details>

Reply:
Thanks for figuring out the current default, I'll change the default value accordingly then.
With regards to the network specific settings, I could also try to add something for those.
Networkd seems to only require adding the field name to those allowed for the network section https://github.com/NixOS/nixpkgs/blob/3d705240031668a009a6944385d2ae21d24d16d6/nixos/modules/system/boot/networkd.nix#L14-L26
For NetworkManager there seems to be no precendent for connection specific settings exposed in nix, only an option for including text in the config file in the `[connection]` section. Adding an "enable mDNS for all connections by default" type setting should be pretty easy though. https://github.com/NixOS/nixpkgs/blob/3d705240031668a009a6944385d2ae21d24d16d6/nixos/modules/services/networking/networkmanager.nix#L153-L170

Reply:
I've checked both the networkd and NetworkManager module more carefully again, and it seems to me like it is already possible for both to enable mdns per network/connection. The section I referenced for networkd above was for the global network options, the per-network one has a field for mdns already.
@raphaelr do you believe this PR needs any more work in that direction? If so, what in particular?
The LLMNR option also does not document the need to enable per-network for it to work, so I also dropped that part when copying the documentation from the [resolved.conf docs](https://www.freedesktop.org/software/systemd/man/resolved.conf.html).
I've changed the default value back to `true` with c99bf5e68c355d6e21c46d77f063ca9d459152e6